src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,68 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.model.Circle;
+import core.basesyntax.model.Figure;
+import core.basesyntax.model.IsoscelesTrapezoid;
+import core.basesyntax.model.Rectangle;
+import core.basesyntax.model.RightTriangle;
+import core.basesyntax.model.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    static final int COLOR_NUMBER = 5;
+    static final int RANDOM_NUMBER = 100;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureIndex = new Random().nextInt(COLOR_NUMBER);
+        switch (figureIndex) {
+            case 0:
+                return createCircle(colorSupplier.getRandomColor(), getRandom());
```

Comment:
```suggestion
                return createRandomCircle();
```
lets better create methods without parameters
because it gives no difference when you create a figure here or in another method with all the same parameters as constructor